### ISSUE TITLE
S3 Region Pattern fails for -fips endpoints

### ISSUE THREAD

**Issue by insdami** [aws/aws-sdk-java#2386]
The [pattern](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/Region.java#L327
) which is used to [get the region](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java#L4641) doesn't work with fips endpoints i.e `s3-fips.us-east-1.amazonaws.com`

## Describe the bug

regex in https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/Region.java#L327 doesn't take into account `*-fips` endpoints

## Expected Behavior

Match on fips endpoints

## Current Behavior

```scala
scala> val authority = URI.create("https://s3-fips.us-east-1.amazonaws.com").getAuthority
authority: String = s3-fips.us-east-1.amazonaws.com

scala> Region.S3_REGIONAL_ENDPOINT_PATTERN.matcher(authority).matches
res16: Boolean = false

scala> val authority = URI.create("https://s3.us-east-1.amazonaws.com").getAuthority
authority: String = s3.us-east-1.amazonaws.com

scala> Region.S3_REGIONAL_ENDPOINT_PATTERN.matcher(authority).matches
res17: Boolean = true
```

## Your Environment

* AWS Java SDK version used: 1.11.793
* JDK version used: 1.8

**Comment by debora-ito** [comment]
Thank you for reporting @insdami, I can reproduce.

**Comment by debora-ito** [comment]
Issue fixed in `1.11.840`. 

Closing this, feel free to reach out  if you have any other question or concern.
